test/cmdlineargs.jl

```diff
@@ -0,0 +1,168 @@
+let exename = joinpath(JULIA_HOME, (ccall(:jl_is_debugbuild, Cint, ()) == 0 ? "julia" : "julia-debug"))
+    # --version
+    let v = split(readall(`$exename -v`), "julia version ")[end]
+        @test VERSION == VersionNumber(v)
+    end
+    @test readall(`$exename -v`) == readall(`$exename --version`)
+
+    # --help
+    @test startswith(readall(`$exename -h`), "julia [options] [program] [args...]")
+    @test startswith(readall(`$exename --help`), "julia [options] [program] [args...]")
+
+    # --quiet
+    # This flag is indirectly tested in test/repl.jl
+
+    # --home
+    @test success(`$exename -H $JULIA_HOME`)
+    @test success(`$exename --home=$JULIA_HOME`)
+
+    # --eval
+    @test  success(`$exename -e "exit(0)"`)
+    @test !success(`$exename -e "exit(1)"`)
+    @test  success(`$exename --eval="exit(0)"`)
+    @test !success(`$exename --eval="exit(1)"`)
+    @test !success(`$exename -e`)
+    @test !success(`$exename --eval`)
+
+    # --print
+    @test readall(`$exename -E "1+1"`) == "2\n"
+    @test readall(`$exename --print="1+1"`) == "2\n"
+    @test !success(`$exename -E`)
+    @test !success(`$exename --print`)
+
+    # --post-boot
+    @test  success(`$exename -P "exit(0)"`)
+    @test !success(`$exename -P "exit(1)"`)
+    @test  success(`$exename --post-boot="exit(0)"`)
+    @test !success(`$exename --post-boot="exit(1)"`)
+    @test !success(`$exename -P`)
+    @test !success(`$exename --post-boot`)
+
+    # --load
+    let testfile = tempname()
+        try
+            open(testfile, "w") do io
+                println(io, "testvar = :test")
+            end
+            @test split(readchomp(`$exename --load=$testfile -P "println(testvar)"`), '\n')[end] == "test"
+            @test split(readchomp(`$exename -P "println(testvar)" -L $testfile`), '\n')[end] == "test"
+        finally
+            rm(testfile)
+        end
+    end
+    # -L, --load requires an argument
+    @test !success(`$exename -L`)
+    @test !success(`$exename --load`)
+
+    # --cpu-target
+    @test !success(`$exename -C invalidtarget`)
```

Comment:
The tests pass with `julia-debug` on OSX, but I get the same error with the debug build if I delete `sys.dylib`.  Removing the `-C target` and `--target=target` allow the tests to pass.

```julia
    * cmdlineargs         exception on 1: ERROR: LoadError: test failed: !(success(@cmd "\$exename -C invalidtarget"))
 in expression: !(success(@cmd "\$exename -C invalidtarget"))
 in error at error.jl:19
 in default_handler at test.jl:25
 in do_test at test.jl:50
 in anonymous at no file:58
 in runtests at /Users/jacobbolewski/Julia/julia/test/testdefs.jl:78
 in anonymous at multi.jl:656
 in run_work_thunk at multi.jl:617
 in remotecall_fetch at multi.jl:690
 in remotecall_fetch at multi.jl:705
 in anonymous at task.jl:1406
while loading cmdlineargs.jl, in expression starting on line 1
ERROR: LoadError: LoadError: test failed: !(success(@cmd "\$exename -C invalidtarget"))
 in expression: !(success(@cmd "\$exename -C invalidtarget"))
 in error at error.jl:19
 in default_handler at test.jl:25
 in do_test at test.jl:50
 in anonymous at no file:58
 in runtests at /Users/jacobbolewski/Julia/julia/test/testdefs.jl:78
 in anonymous at multi.jl:656
 in run_work_thunk at multi.jl:617
 in remotecall_fetch at multi.jl:690
 in remotecall_fetch at multi.jl:705
 in anonymous at task.jl:1406
while loading cmdlineargs.jl, in expression starting on line 1
while loading /Users/jacobbolewski/Julia/julia/test/runtests.jl, in expression starting on line 3
```